vignettes/spatial_dhmv_query.Rmd

```diff
@@ -0,0 +1,490 @@
+
+How about **[QGIS](https://qgis.org)?** *"Spatial Without Compromise"*, they advertise. Unfortunately, that program also was unable to open the file downloaded in R.
+
+However, QGIS can actually connect to WCS directly. This does not help much if you need the data in R. Yet, why not.
+
+Look and behold: QGIS can actually connect to <`https://geo.api.vlaanderen.be/DHMV/wcs`> and query elevation data for Flanders; good indication that the WCS is, somehow, functional. I could zoom and move, export images. I could even export the data. I stopped exporting the data when it threatened to entirely fill my tiny system partition (the raster data is more than 100GB) - good confirmation that we want web services for this.
+
+And, luckily, at some careless map movement, I received an error output from QGIS. With it, QGIS gave me a working URL to query DHMV: 
+
+-  `https://geo.api.vlaanderen.be/DHMV/wcs?SERVICE=WCS&VERSION=1.0.0&REQUEST=GetCoverage&FORMAT=GeoTIFF&COVERAGE=DHMVII_DTM_1m&BBOX=162064,165735,170953,168882&CRS=EPSG:31370&RESPONSE_CRS=EPSG:31370&WIDTH=1622&HEIGHT=574`
+
+
+```{r echo=FALSE, out.width='90%'}
+knitr::include_graphics("../man/figures/qgis_lead.avif")
+```
+
+Let's dissect this URL.
+
+
+<a id="org803f24d"></a>
+
+# The Solution: Query Building
+
+If you start at the question mark of the URL, and split at every ampersand, you can extract the following components from the (correct) QGIS URL:
+
+| **correct:**           |                                   |
+|---------------------- |--------------------------------- |
+| SERVICE                | WCS                               |
+| VERSION                | 1.0.0                             |
+| REQUEST                | GetCoverage                       |
+| FORMAT                 | GeoTIFF                           |
+| COVERAGE               | DHMVII_DTM_1m |
+| BBOX                   | 162064,165735,170953,168882       |
+| CRS                    | EPSG:31370                        |
+| RESPONSE_CRS           | EPSG:31370                        |
+| WIDTH                  | 1622                              |
+| HEIGHT                 | 574                               |
+
+Compare this to the (unsuccessful) `ows4R` attempt:
+
+| **non-functional:** |                                   |
+|------------------- |--------------------------------- |
+| service             | WCS                               |
+| version             | 2.0.1                             |
+| coverageId          | DHMVII_DTM_1m |
+| subset              | x(148599,148601)                  |
+| subset              | y(208899,208901)                  |
+| format              | image/tiff                        |
+| request             | GetCoverage                       |
+
+You can selectively shuffle and adjust components, generalizing the working query and learning about the WCS API. Which feels a bit like [fuzzing](https://en.wikipedia.org/wiki/Fuzzing) the `geo.api`.
```

Comment:
'geo.api' confused me a bit here (hence so does the geo.api subdomain). I was confusing with the recent [OGC API](https://ogcapi.ogc.org/), i.e. the successor of OGC Web Service Standards.  IIUC, OGC API has dedicated server/client implementations in Java and Python, called the OGC "[GeoAPI](https://www.ogc.org/publications/standard/geoapi/)" standard, and is still evolving. The OGC API offers modern Web API approaches compared to the (legacy!) OGC Web Service Standards (which will probably remain in place for a long time).

Yet we shouldn't confuse with OGC API / GeoAPI since we're dealing with web services, not OGC API.

BTW, interacting with OGC API is provided by GDAL ([vector](https://gdal.org/en/latest/drivers/vector/oapif.html) / [raster](https://gdal.org/en/latest/drivers/raster/ogcapi.html)), which is the obvious gateway for R packages like {sf} and {terra}, as these already interface GDAL.

(Also, the webservice standards are supported by GDAL, hence should be able to use with GDAL-interfacing R packages)

The term 'geo.api' pops up a couple of times in the vignette so maybe you can improve or replace it (maybe 'WCS request' or something).

Reply:
Interesting and good to know! I have replaced all "API" by WCS or interfaces, except when writing about the `geo.api.vlaanderen.be`.

Reply:
Thanks!